File: TestBinarySearch/Python/TestBinarySearch.py
#TestBinarySearch
#19/02/2021

#binarySearch(list, element)
#Description: The function return the position for a element of a list.
def binarySearch(list, element):
	minor = 0
	mayor = len(list) - 1
	
	while minor <= mayor:
		medium = round((minor + mayor)/2)
		estimated = list[medium]
		if estimated == element:
			return medium
		if estimated > element:
			mayor = medium
		else:
			minor = medium + 1
	return "Not found"

#TEST	
myList = [1, 3, 5, 7, 9]
print(binarySearch(myList, 9))
# print(binarySearch(myList, -1)) # The function brokes searching negative numbers
	

File: TestBinarySearch/C/TestBinarySearch.c
//TestBinarySearch.c
//23/02/2021
#include <stdio.h>
#include <math.h>

double BinarySearch(double list[], int listSize, double element){
	
	int medium = 0;
	int minor = 0;
	int mayor = listSize - 1;
	double estimated = 0;
	
	while(minor <= mayor){
		medium = minor + (mayor - minor)/2;
		estimated = list[medium];
		if(estimated == element){
			return medium;
		}
		if(estimated < element){
			minor = medium + 1;
		}
		else{
			mayor = medium - 1;
		}	
	}
		return -1;
};

int main(){
	
	double list[5] = {1, 3, 5, 7, 9};
	double pos;
	double element = 9;
	pos = BinarySearch(list, sizeof(list)/sizeof(double),  element);
	printf("Element is in position %f", pos);
	
	return 0;
}
